test_util/src/lsp.rs

```diff
@@ -122,7 +122,16 @@ where
   R: de::DeserializeOwned,
 {
   let maybe_params = match maybe_params {
-    Some(params) => Some(serde_json::from_value(params)?),
+    Some(params) => {
+      Some(serde_json::from_value(params.clone()).map_err(|err| {
+        anyhow::anyhow!(
+          "Could not deserialize message '{}': {}\n\n{:?}",
+          method,
+          err,
+          params
+        )
+      })?)
+    }
```

Comment:
This was to get better information while debugging so I left it in because it's probably useful in the future.